plugin/loader/loader.go

```diff
@@ -81,41 +84,83 @@ func (loader *PluginLoader) Initialize() error {
 	return nil
 }
 
-//Run the plugins
-func (loader *PluginLoader) Run() error {
+// Inject hooks all the plugins into the appropriate subsystems.
+func (loader *PluginLoader) Inject() error {
 	for _, pl := range loader.plugins {
-		switch pl := pl.(type) {
-		case plugin.PluginIPLD:
-			err := runIPLDPlugin(pl)
+		if pl, ok := pl.(plugin.PluginIPLD); ok {
```

Comment:
I assume this is because we want plugins to be able to implement more that one interface, right?

Reply:
Yeah, there's no reason for these interfaces to be exclusive.